### PR TITLE
DEV-47: Update to PHP 8.2

### PR DESCRIPTION
**Pull request by byrond**
https://palantir.atlassian.net/browse/DEV-47

Update DDEV and CircleCI configs to use PHP 8.2

~Open question: Do we need to update the `composer.json` minimum PHP version and readme?~ Yes

## Testing

- Follow the steps in the readme to create a new project using this branch (update-php-8.2). For the Composer command use:

`composer create-project palantirnet/drupal-skeleton example dev-update-php-8.2 --no-interaction`

### CODE REVIEW

**Comment by becw** [comment]
We talked about this in last week's devops meeting, and the main blocker is Drupal contrib compatibility. I don't think we have a clear view of what is and isn't compatible with PHP 8.2 yet, though -- we're in the middle of an update on a big site and that should flag any critical modules that would block this update. So, we'll hopefully be able to move on this by September.

File: README.md [hunk 1]
```diff
@@ -12,8 +12,9 @@ You should have the development dependencies installed on your Mac before you be
 
 The development dependencies are:
 
-* PHP 8.1+
+* PHP 8.2+
   * Check your PHP version from the command line using `php --version`
+  * If your project cannot support PHP 8.2, you should use PHP 8.1.18.
```
Comment (byrond) [inline README.md]: Do we still need this?

Comment (becw) [reply README.md]: Nope, I don't think so!

Comment (byrond) [reply README.md]: Removed!

**Review by becw — APPROVED** [review]

**Pull request merged**